.github/workflows/reposchutz.yml

```diff
@@ -7,20 +7,20 @@ jobs:
   check:
     name: Protection checks
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
     timeout-minutes: 5
     env:
       GIT_DIR: git-dir.git
       HEAD_SHA: ${{ github.event.pull_request.head.sha }}
       BASE_SHA: ${{ github.event.pull_request.base.sha }}
 
     steps:
-      - name: Create work area
-        run: |
-          set -x
-          git init -b main
-          # git fetch --shallow-exclude="${BASE_SHA}" origin "${BASE_SHA}"  # GitHub doesn't support this
-          git remote add origin "${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}"
-          git fetch origin "${BASE_SHA}" "${HEAD_SHA}"
+      - name: Clone main branch
+        uses: actions/checkout@v2
+
+      - name: Fetch PR commits
+        run: git fetch origin "${BASE_SHA}" "${HEAD_SHA}"
```

Comment:
No.

Because this workflow runs in `pull_request_target`, the branch that gets checked out is the target branch (ie: `main`).  At this point we don't have the PR, so we need to explicitly fetch it.

The key point here is that we run the scripts that are on the `main` branch, not the (possibly modified) modified ones from the proposed branch.  That's for security reasons (although this workflow has zero permissions aside from reading from a public repository and deciding a +1/-1 check result).  Of course, this means that we can't ever make changes to the scripts and update node modules at the same time, but I don't think we ever do that...

Reply:
Right, sorry for the noise!